### ISSUE TITLE
Items jump out of network nodes by themselves.

### ISSUE THREAD

**Issue by Boomer-1** [Slimefun/Slimefun4#2600]
## :round_pushpin: Description (REQUIRED)

I have had Zinc Dust jump magically out of the output nodes a stack at a time and hit the ground until they despawn. It also happened with magnesium dust. I have recorded a video of it happening with nuclear reactor coolant cells doing the same thing. The dust came out of fluffy barrels, where the coolant came from freezers.

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

Setup a freezer to turn blue ice into reactor coolant cells by only allowing blue ice in from the output node, and reactor cells coming out from the input node to go to the nuclear power plant. After some time, they start randomly falling on the floor.

Here's the video of it happening

https://youtu.be/VKLngR1QHIo

## :bulb: Expected behavior (REQUIRED)

The correct behavior would be to stay in the freezer if there is no room to go anywhere else. Then the freezer should stop producing coolant cells if 2 stacks are in the right side of the freezer.

## :scroll: Server Log

there are no errors in the log. 
[latest (1).log](https://github.com/Slimefun/Slimefun4/files/5648274/latest.1.log)

## :open_file_folder: /error-reports/ Folder

the folder was empty

## :compass: Environment (REQUIRED)

 - Server Software (Spigot/Paper):
 - Minecraft Version: 1.16.3
 - Slimefun Version:
 - CS-CoreLib Version:
![image](https://user-images.githubusercontent.com/75179680/101271388-d58e2380-3747-11eb-98f4-03da2f0998c5.png)
![image](https://user-images.githubusercontent.com/75179680/101271392-e474d600-3747-11eb-8ab1-dd046be81293.png)

**Comment by TheBusyBiscuit** [comment]
That is completely intended.
It happens when an Inventory is blocked by the time the item arrives there.

You can also disable it in the config but note that items that could not be distributed properly will then be voided.

**Comment by TheBusyBot** [comment]
Your issue has been closed by an admin, it may fall under one or more of the following categories.
**Please wait for an admin to tick off the points that apply.**
<hr>

* [ ] You did not follow our template. Please follow the Issue template to help us identify your issue more effectively.
* [ ] You did not provide any information about your versions (We absolutely need the exact version numbers that are installed on your Server, \"latest\" is not helpful)
* [ ] You did not provide a proper description to the problem. Try to write at least 4-6 sentences and/or provide screenshots or videos on how to reproduce this.
* [ ] We were unable to reproduce issue, if you think your issue still persists then please comment down below and give a better description on how to reproduce it.
* [ ] Your issue was posted in a foreign language, we only accept english issues on here.
* [ ] Your issue is not a bug, it is intended to work this way.
* [ ] Your issue is not really a bug, it is a limitation or simply something we have no control over.
* [ ] Your issue is not a bug, please only use this issue tracker to report bugs. Any other kind of communication should happen on discord.
* [ ] Your issue has already been reported before, it is a duplicate. Check the other issues first before posting!
* [ ] You posted an error without using pastebin. Please always post errors via pastebin otherwise they become nearly unreadable.
* [ ] You seem to be reporting multiple bugs at once. Please make a separate issue for each bug you encountered, so we can properly handle them individually.
* [ ] Your issue has already been fixed in a later version of Slimefun or CS-CoreLib, you should update.
* [ ] You are using an outdated and unsupported version of Slimefun / CS-CoreLib, again, you should update.
* [ ] You are using an unofficially modified build of Slimefun. We only support official versions of Slimefun - for obvious reasons.
* [ ] You are using an unsupported version of Minecraft. We only provide support for the Minecraft versions Slimefun was developed for, older versions are not supported anymore.
* [ ] You are using a \"stable\" version of Slimefun (prefixed with \"RC - \"), your issue may have been fixed in a development build, so we only accept bug reports from those.
* [ ] You are on the wrong issue tracker. We would like to remind you that this Issue Tracker is **only for Slimefun**. To report bugs on any addons, head to the corresponding issue tracker of that addon.
<hr>

Please respond below, if you have any further questions.
Do **not** open a new Issue unless explicitly told otherwise, comment below or edit your post instead.<br>
Make sure to check out our article on [How to report bugs](https://github.com/Slimefun/Slimefun4/wiki/How-to-report-bugs) for even more information.

**Comment by Boomer-1** [comment]
If this is intentional then it is a bug. It does not happen consistently. the three freezers in the video below the one that were shooting out items, has NEVER HAPPENED when they hit 100% full. Nor have any of the 100 ore grinders i have, 40 gold pans or 20 dust washers ever tossed any items out of their inventories. I had 2 fluffy barrels doing this. One was less than 10% full, another was about 60% full. I have tons of chests 100% full, and none of them in 2 years of playing this game have ever done what you said is intentional. This is a bug. If it was intentional, then it is bugged because it doesn't happen 99% of the time. I guess I don't understand why it is even in the game to begin with. It makes zero sense to have it happen. Why should i expect to see thousands of my items lying on the ground because they are being kicked out of the system because it's full. why not just shut the machine off until there is room?

**Comment by TheBusyBiscuit** [comment]
It is not a bug, it's absolutely intentional.
But like I said: You can turn it off in the config.

> the three freezers in the video below the one that were shooting out items, has NEVER HAPPENED when they hit 100% full. Nor have any of the 100 ore grinders i have, 40 gold pans or 20 dust washers ever tossed any items out of their inventories. I had 2 fluffy barrels doing this. One was less than 10% full, another was about 60% full. I have tons of chests 100% full

I feel like you don't really understand. It has nothing to do with the "fullness" of inventories.
Cargo networks check if there is a room for an item. If there is, then they sent out an item.
But if - for whatever reason - the inventory was modified inbetween these two events (e.g. asynchronous machine access, energy networks, other cargo networks, hoppers and countless other ways this could interefere due to the multi-threaded nature of inventories), then the item gets deleted.
Since a few updates there is now a deliberate check to drop items affected by this bug, although it can be toggled in the config.

> none of them in 2 years of playing this game have ever done what you said

Like I said, these items used to be just deleted from existence. But people have complained about cargo networks deleting their items for years.

> why not just shut the machine off until there is room?

Normally this happens, but due to unforeseeable events it can sometimes happen to leak out items.
If it bothers you, simply toggle it in the config.

**Comment by TheBusyBiscuit** [comment]
And I am aware that this is not an ideal scenario, it absolutely is nothing more than a compromise.
But cargo networks are simply completely faulty and broken in of itself and the entire cargo system is beyond repair, it needs to be redone completely eventually.
But until someone makes a new cargo system, compromises like these are simply there to accept.
